src/shogun/regression/LinearRidgeRegression.cpp

```diff
@@ -75,24 +73,19 @@ bool CLinearRidgeRegression::train_machine(CFeatures* data)
 	for (int32_t i=0; i<num_feat; i++)
 		kernel_matrix.matrix[i+i*num_feat]+=m_tau;
 
-	for (int32_t i=0; i<num_vec; i++)
-	{
-		SGVector<float64_t> v = feats->get_feature_vector(i);
-		ASSERT(v.vlen==num_feat)
-
-		cblas_dger(CblasColMajor, num_feat,num_feat, 1.0, v.vector,1,
-				v.vector,1, kernel_matrix.matrix, num_feat);
+	Map<MatrixXd> eigen_kernel_matrix(kernel_matrix.matrix, num_feat,num_feat);
+	Map<MatrixXd> eigen_feats_matrix(feats_matrix.matrix, num_feat,num_vec);
+	Map<VectorXd> eigen_y(y.vector, num_feat);
+	Map<VectorXd> eigen_labels(((CRegressionLabels*)m_labels)->get_labels(),num_vec);
 
-		cblas_daxpy(num_feat, ((CRegressionLabels*) m_labels)->get_label(i), v.vector, 1, y.vector, 1);
+	eigen_kernel_matrix += eigen_feats_matrix*eigen_feats_matrix.transpose();
```

Comment:
looks good.
Potential for linalg operations, i.e. covariance matrix computation.
If you are up for this, feel free to open an issue to add that to linalg